Storage/src/Connection/Rest.php

```diff
@@ -304,6 +304,28 @@ public function insertObject(array $args = [])
             ]
         ];
 
+        // Passing on preconditions if present
+        foreach (RetryTrait::$condIdempotentOps['objects.insert'] as $condition) {
```

Comment:
After looking at it again, I think we should move this logic to `resolveUploadOptions`. There is some customisation of arguments already happening in that method and it is only applicable to uploads as well.

Moving will ensure this method stays clean like it was before.
WDYT?

Reply:
```suggestion
        // The preconditions we want to extract out of arguments
        // into our query params.
        $preconditions = RetryTrait::$condIdempotentOps['objects.insert'];
        foreach ($preconditions as $condition) {
```

Reply:
Yeah. So we can extact the uploaderOptions updation logic in `resolveUploadOptions()` but the logic to update the query params needs to stay in `insertObject()` as `$uriParams`, thus query params are initialised here. Did the same in [3c7e01b](https://github.com/googleapis/google-cloud-php/pull/5730/commits/3c7e01b39320ada00e922a0095d5d2663948e668)

On thinking, I observed that we do not need to send preconditions / retry strategy to `$args['uploaderOptions']` as they are just required by the custom retry callback, so we can simple send $args instead of $args['uploaderOptions'] below:

Storage > src > Connection > Rest.php > resolveUploadOptions : line 373 to 378 in commit [3c7e01b](https://github.com/googleapis/google-cloud-php/pull/5730/commits/3c7e01b39320ada00e922a0095d5d2663948e668)
```php
$retryFunc = $this->getRestRetryFunction(
    'objects',
    'insert',
-    $args['uploaderOptions']
+    $args
);
```